aws_doc_sdk_examples_tools/snippets.py

```diff
@@ -213,10 +214,41 @@ def collect_snippet_files(
                                 file=snippet_file,
                                 line_start=0,
                                 line_end=len(code),
-                                code="".join(code),
+                                code="".join(
+                                    strip_snippet_tags(strip_spdx_header(code))
+                                ),
                             )
 
 
+def strip_snippet_tags(lines: List[str]) -> List[str]:
+    return [line for line in lines if not has_snippet_tag_or_spdx_header(line)]
+
+
+def strip_spdx_header(file: List[str]) -> List[str]:
+    has_copyright = (
+        re.match(
+            validator_config.SPDX_LEADER + validator_config.SPDX_COPYRIGHT, file[0]
+        )
+        is not None
+    )
+    has_license = (
+        re.match(validator_config.SPDX_LEADER + validator_config.SPDX_LICENSE, file[1])
+        is not None
+    )
+    has_spacer = file[2] == ""
```

Comment:
Is`file` a list of lines? If so, isn't `file[2]` making an assumption about the number of lines in the leader and license?

Reply:
Done